web/error.go

```diff
@@ -0,0 +1,184 @@
+package web
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"darvaza.org/core"
+	"darvaza.org/x/web/consts"
+)
+
+var (
+	_ Error            = (*HTTPError)(nil)
+	_ core.Unwrappable = (*HTTPError)(nil)
+	_ http.Handler     = (*HTTPError)(nil)
+)
+
+// Error is an error that knows its HTTP Status Code
+type Error interface {
+	Error() string
+	HTTPStatus() int
+}
+
+// HTTPError extends [core.WrappedError] with HTTP Status Code
+type HTTPError struct {
+	Err  error
+	Code int
+	Hdr  http.Header
+}
+
+// HTTPStatus returns the HTTP status code associated with the Error
+func (err *HTTPError) HTTPStatus() int {
+	switch {
+	case err.Code == 0:
+		return http.StatusOK
+	case err.Code < 0:
+		return http.StatusInternalServerError
+	default:
+		return err.Code
+	}
+}
+
+// Header returns a [http.Header] attached to this error for custom fields
+func (err *HTTPError) Header() http.Header {
+	if err.Hdr == nil {
+		err.Hdr = make(http.Header)
+	}
+	return err.Hdr
+}
+
+// AddHeader appends a value to an HTTP header entry of the HTTPError
+func (err *HTTPError) AddHeader(key, value string) {
+	err.Header().Add(key, value)
+}
+
+// SetHeader sets the value of a header key of the HTTPError
+func (err *HTTPError) SetHeader(key, value string) {
+	err.Header().Set(key, value)
+}
+
+// DeleteHeader removes a header key from the HTTPError if present
+func (err *HTTPError) DeleteHeader(key string) {
+	err.Header().Del(key)
+}
+
+// ServeHTTP is a very primitive handler that will try to pass the error
+// to a [middleware.ErrorHandlerFunc] provided via the request's context.Context
```

Comment:
Here I miss the continuation of the sentence, `will try to .....` 

+`if that exists, if not it calls the ``Render`` function`